backend/kesaseteli/applications/jobs/daily/tests/test_clean_youth_applications.py

```diff
@@ -0,0 +1,166 @@
+        application.created_at = created_at
+        application.save()
+    return applications
+
+
+@pytest.mark.django_db
+def test_clean_inactive_youth_applications_older_than_week():
+    date_week_ago = timezone.now() - relativedelta(weeks=1)
+    date_less_than_week_ago = timezone.now() - relativedelta(days=6)
+
+    # Create different applications
+    inactive_applications_created_week_ago = set_creation_date(
+        InactiveYouthApplicationFactory.create_batch(size=5),
+        created_at=date_week_ago,
+    )
+
+    inactive_applications_created_less_than_week_ago = set_creation_date(
+        InactiveYouthApplicationFactory.create_batch(size=5),
+        created_at=date_less_than_week_ago,
+    )
+    active_applications_created_week_ago = set_creation_date(
+        ActiveYouthApplicationFactory.create_batch(size=5),
+        created_at=date_week_ago,
+    )
+
+    # Run the batch job
+    clean_inactive_youth_applications.Job().execute()
+
+    assert not YouthApplication.objects.filter(
+        pk__in=[app.id for app in inactive_applications_created_week_ago]
+    ).exists(), "Inactive applications should have been cleaned"
+
+    assert YouthApplication.objects.filter(
+        pk__in=[app.id for app in inactive_applications_created_less_than_week_ago]
+    ).exists(), "Newer inactive applications should still exist"
+
+    assert YouthApplication.objects.filter(
+        pk__in=[app.id for app in active_applications_created_week_ago]
+    ).exists(), "Active applications should still exist"
+
+
+@pytest.mark.django_db
+def test_clean_rejected_youth_applications_older_than_year():
+    date_year_ago = timezone.now() - relativedelta(year=1)
+    date_less_than_year_ago = timezone.now() - relativedelta(years=1, days=-1)
+
+    # Create different applications
+    rejected_applications_created_year_ago = set_creation_date(
+        RejectedYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    rejected_applications_created_less_than_year_ago = set_creation_date(
+        RejectedYouthApplicationFactory.create_batch(size=5),
+        created_at=date_less_than_year_ago,
+    )
+
+    unhandled_applications_created_year_ago = set_creation_date(
+        UnhandledYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    accepted_applications_created_year_ago = set_creation_date(
+        AcceptedYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    # Run the batch job
+    clean_rejected_youth_applications.Job().execute()
+
+    assert not YouthApplication.objects.filter(
+        pk__in=[app.id for app in rejected_applications_created_year_ago]
+    ).exists(), "Rejected application older than year should have been cleaned"
+
+    assert YouthApplication.objects.filter(
+        pk__in=[app.id for app in rejected_applications_created_less_than_year_ago]
+    ).exists(), "Newer rejected applications should still exist"
```

Comment:
This won't assert if e.g. a single application from `rejected_applications_created_less_than_year_ago` has been deleted (See [exists](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.exists) documentation).

Reply:
That's true! Perhaps I could check that the length of the result is same as the length of `rejected_applications_created_less_than_year_ago`

Reply:
Well, I ended up reverting the for-loop for the exists checks